### PR TITLE
Added jax.numpy.quantile frontend and tests

### LINKED ISSUE #19702: quantile

### PR DESCRIPTION
**Pull request by illia-bab**
Close #19702

### CODE REVIEW

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
As there has been a little change to the testing logic. Kindly add the arguments which I mentioned below. 
Also your test is failing for `paddle`, I have added the falsifying example below so you can easily reproduce the failure, Thanks! :slightly_smiling_face:  

```
E               AssertionError: returned shape = (), ground-truth returned shape = (1,)
E               Falsifying example: test_jax_quantile(
E                   on_device='cpu',
E                   frontend='jax',
E                   backend_fw='paddle',
E                   dtype_array_axes_q=(['float64'], [array([-1.])], 0, array([0.5])),
E                   overwrite_input=False,
E                   method='linear',
E                   keepdims=False,
E                   test_flags=FrontendFunctionTestFlags(
E                       num_positional_args=2,
E                       with_out=False,
E                       inplace=False,
E                       as_variable=[False],
E                       native_arrays=[False],
E                       generate_frontend_arrays=False,
E                   ),
E                   fn_tree='ivy.functional.frontends.jax.numpy.quantile',
E               )
E               
E               You can reproduce this example by temporarily adding @reproduce_failure('6.82.0', b'AXicY2BkAAMoBaTBrAMMKAAAC9MAxQ==') as a decorator on your test case

ivy_tests/test_ivy/helpers/assertions.py:77: AssertionError
```

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hey @illia-bab ,

Great changes :+1: 
I've tested it locally there seems to be inconsistency with the `numpy` backend as well, I'll be providing the falsifying example below so it can be easier for you, Thanks! :slightly_smiling_face:

```
E       AssertionError:  the results from backend numpy and ground truth framework jax do not match
E        [[1.]]!=[[0.5]] 
E       
E       
E       Falsifying example: test_jax_quantile(
E           on_device='cpu',
E           frontend='jax',
E           backend_fw='numpy',
E           dtype_array_axes_q=(['float32'], [array([[1. ],
E                    [1. ],
E                    [0.5],
E                    [0.5]], dtype=float32)], 0, array([0.5])),
E           overwrite_input=False,
E           method='nearest',
E           keepdims=False,
E           test_flags=FrontendFunctionTestFlags(
E               num_positional_args=2,
E               with_out=False,
E               inplace=False,
E               as_variable=[False],
E               native_arrays=[False],
E               generate_frontend_arrays=False,
E           ),
E           fn_tree='ivy.functional.frontends.jax.numpy.quantile',
E       )
E       
E       You can reproduce this example by temporarily adding @reproduce_failure('6.82.0', b'AXicY2BkYGZkQAaMqMwDCO4BVDVwLguIAABAcgJN') as a decorator on your test case
```

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_statistical.py [hunk 1]
```diff
@@ -1185,3 +1185,81 @@ def test_jax_cov(
...
+
+    return array_dtype, array, axes, q
+
+
+@handle_frontend_test(
+    fn_tree="jax.numpy.quantile",
+    dtype_array_axes_q=_get_array_axes_probs(),
+    overwrite_input=st.just(False),
+    keepdims=st.booleans(),
+    method=st.sampled_from(["linear", "lower", "higher", "midpoint", "nearest"]),
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_statistical.py:1237]: `nearest_jax` would be passed in here except for `nearest`

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hey @illia-bab , 

Great changes, just a few more changes I've mentioned below :slightly_smiling_face:  
To make it more cleaner as we discussed earlier, what we can do is we can add the `mode` in interpolation naming it as `nearest_jax`, while keeping the `nearest` also there, the main takeaway would be, whenever we run a `jax` frontend test, we would need the `nearest_jax` rather than `nearest` to replicate jax's behaviour in every backend as well. The main changes here would be as follows: 


1. Remove the `nearest_jax` argument from here and add it as a mode of `interpolation` in every backend removing the separate argument which we made, adding the functionality as an interpolation mode naming `nearest_jax`  
2. You've missed adding the docstring of the mode in the original ivy function definition https://github.com/unifyai/ivy/blob/951f116ce4d473de5ab6d97312dac2eaf369a757/ivy/functional/ivy/experimental/statistical.py#L284
3. In the frontend test you have to pass in the  `nearest_jax` mode to test jax's functionality :+1:

Let me know if you have any other questions, thanks! :slightly_smiling_face:

File: ivy/functional/frontends/jax/numpy/statistical.py [hunk 2]
```diff
@@ -486,3 +486,25 @@ def cov(m, y=None, rowvar=True, bias=False, ddof=None, fweights=None, aweights=N
...
+    /,
+    *,
+    axis=None,
+    out=None,
+    overwrite_input=False,
+    method="linear",
+    keepdims=False,
+    interpolation=None,
+):
+    return ivy.quantile(
```
Comment (zaeemansari70) [inline ivy/functional/frontends/jax/numpy/statistical.py:508]: We can handle this by adding a simple condition here that if the mode is `nearest`, we can pass in `nearest_jax` to the ivy function
```
if mode=="nearest":
    ivy.quantile(    a, q, axis=axis, keepdims=keepdims, interpolation='nearest_jax', out=out
    )
```

Thanks! :slightly_smiling_face:

**Comment by illia-bab** [comment]
We're ready to merge!)

**Pull request merged**